.github/workflows/haskell.yml

```diff
@@ -21,6 +21,22 @@ jobs:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ matrix.cabal }}
 
+    - uses: actions/cache@preview
+      name: Cache ~/.cabal/packages
+      with:
+        path: ~/.cabal/packages
+        key: ${{ runner.os }}-${{ matrix.ghc }}-cabal-packages-${{ hashFiles('**/fused-effects.cabal') }}
+    - uses: actions/cache@preview
+      name: Cache ~/.cabal/store
+      with:
+        path: ~/.cabal/store
+        key: ${{ runner.os }}-${{ matrix.ghc }}-cabal-store-${{ hashFiles('**/fused-effects.cabal') }}
+    - uses: actions/cache@preview
+      name: Cache dist-newstyle
+      with:
+        path: dist-newstyle
+        key: ${{ runner.os }}-${{ matrix.ghc }}-fused-effects-dist-${{ hashFiles('**/fused-effects.cabal') }}
```

Comment:
What’s the significance of including the hash of the cabal file in the key? I get why we’d want to include the OS & ghc version, but I think the cabal file shouldn’t invalidate the cache, but rather update it.

Reply:
We can take this out for sure. Most of the [examples](https://github.com/actions/cache/blob/master/examples.md) show basing the key off of some sort of lock file and invalidating the cache when dependencies change.

Reply:
I guess we could do a `cabal freeze` in CI, hash that, and then delete the freeze file. What do you think?

Reply:
Just removed the hash of the cabal file entirely in e0479d86bbb213e9852cfd0f0b647d7a0aedd3aa, let's see what it does!